packages/atomic/src/components/atomic-search-interface/atomic-search-interface.tsx

```diff
@@ -257,18 +255,19 @@ export class AtomicSearchInterface {
   }
 
   private updateHash() {
-    this.updatingHash = true;
-    defer(() => (this.updatingHash = false));
-    const hash = this.urlManager.state.fragment;
-    window.location.hash = hash;
-  }
-
-  private onHashChange = () => {
-    if (this.updatingHash) {
+    if (this.urlManager.hasFragment(this.fragment)) {
       return;
     }
 
-    this.urlManager.synchronize(this.hash);
+    history.pushState(
```

Comment:
pushState does not trigger a hashChange event
https://developer.mozilla.org/en-US/docs/Web/API/History/pushState